vendor/bat-native-ledger/src/bat/ledger/internal/bitflyer/bitflyer_authorization.cc

```diff
@@ -24,6 +24,22 @@ using std::placeholders::_4;
 namespace ledger {
 namespace bitflyer {
 
+namespace {
+std::string GetEventKeyForLinkingResult(type::Result result) {
+  switch (result) {
+    case type::Result::DEVICE_LIMIT_REACHED:
+      return log::kDeviceLimitReached;
+    case type::Result::MISMATCHED_PROVIDER_ACCOUNTS:
+      return log::kMismatchedProviderAccounts;
+    case type::Result::REQUEST_SIGNATURE_VERIFICATION_FAILURE:
+      return log::kRequestSignatureVerificationFailure;
+    default:
+      NOTREACHED();
+      return "";
+  }
+}
+}  // namespace
```

Comment:
It'd be nice if we could share this implementation among the providers, if possible.

Reply:
Sure!